.github/workflows/flake8.yml

```diff
@@ -19,9 +19,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Check out source repository
-        uses: actions/checkout@v5
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
```

Comment:
what is the advantage of pinning this to a specfic SHA (commit?)

Reply:
This allows us to fix the action version to a known good one, and only change them intentionally (dependabot will suggest updates anyway). Tag references are movable, and indeed are moved all the time (e.g. v5 is a generic tag that gets moved each time a minor 5.x release is made). While using tagged versions can be convenient and should generally be ok, using pinned SHA versions avoids any funny business like what happened in the tj-actions case a few months ago, to cite one example from memory.

This specific `sha # version` format should be supported by dependabot, which should propose updates to newer versions maintaining this format. I will verify this at the next occurrence.